client/src/main/java/io/pravega/client/stream/Transaction.java

```diff
@@ -70,9 +70,11 @@
      * Send a transaction heartbeat and increase transaction's timeout by lease amount of milliseconds.
      *
      * @param lease Additional amount of time in milliseconds by which to increase transaction's timeout.
+     * @param switchOver Boolean indicating whether the previous successful ping request was performed on a different
+     *                   controller process by the client.
      * @throws PingFailedException Ping failed.
      */
-    void ping(long lease) throws PingFailedException;
+    void ping(long lease, boolean switchOver) throws PingFailedException;
```

Comment:
Client is supposed to send periodic ping requests to controller, and not the application. Although this is not yet implemented, we have an Issue #863 for it (which in turn depends on #1034 that has an active PR).

Once we fix #863, client knows when it switches over sending ping requests from unreachable controller instance to another reachable controller instance. Hence, client can set the value of `switchOver` parameter appropriately.